src/components/reusables/ActionBar/ActionBar.test.cy.tsx

```diff
@@ -0,0 +1,153 @@
+    cy.mount(expandableActionBar({ actions: actionButton, open: true }));
+    cy.get('[data-testid="action-bar-icon"]').should('exist');
+    cy.get('button').should('contain', 'Title');
+    cy.get('button').should('contain', 'Subtitle');
+    cy.get('div').should('contain', 'AdditionalText');
+    cy.get('button').should('contain', 'Action');
+    cy.get('div').should('contain', 'Content');
+  });
+
+  it('should render correctly when all props are it is non expandable', () => {
+    const actionButton = <Button>Action</Button>;
+    cy.mount(nonExpandableActionBar({ actions: actionButton }));
+    cy.get('[data-testid="action-bar-icon"]').should('not.exist');
+    cy.get('button').should('contain', 'Title');
+    cy.get('button').should('contain', 'Subtitle');
+    cy.get('div').should('contain', 'AdditionalText');
+    cy.get('button').should('contain', 'Action');
+    cy.get('div').should('not.contain', 'Content');
+  });
+
+  it('non expandable version should not render chevron when it is rendered ', () => {
+    cy.mount(nonExpandableActionBar());
+    cy.get('[data-testid="action-bar-icon"]').should('not.exist');
+  });
+
+  it('should not render chevron when non expandable versjon is rendered ', () => {
+    cy.mount(nonExpandableActionBar());
+    cy.get('[data-testid="action-bar-icon"]').should('not.exist');
+  });
+
+  it('should call handleClick when expandable ActionBar is clicked by mouse', () => {
+    const handleClick = () => {
+      cy.stub();
+    };
+
+    const handleClickSpy = cy.spy(handleClick).as('handleClickSpy');
+
+    cy.mount(expandableActionBar({ onClick: handleClickSpy }));
+
+    cy.get('[data-testid="action-bar"]').click();
+    cy.get('@handleClickSpy').should('have.been.called');
+  });
+
+  it('should call handleClick when expandable ActionBar is clicked by space', () => {
+    const handleClick = () => {
+      cy.stub();
+    };
+    const handleClickSpy = cy.spy(handleClick).as('handleClickSpy');
+
+    cy.mount(expandableActionBar({ onClick: handleClickSpy }));
+
+    cy.get('[data-testid="action-bar"]').type(' ');
+    cy.get('@handleClickSpy').should('have.been.called');
+  });
+
+  it('should call handleClick when expandable ActionBar is clicked by enter', () => {
+    const handleClick = () => {
+      cy.stub();
+    };
+    const handleClickSpy = cy.spy(handleClick).as('handleClickSpy');
+
+    cy.mount(expandableActionBar({ onClick: handleClickSpy }));
+
+    cy.get('[data-testid="action-bar"]').type('{enter}');
+    cy.get('@handleClickSpy').should('have.been.called');
+  });
+
+  it('should not call handleClick when non expandable ActionBar is clicked', () => {
+    const handleClick = () => {
+      cy.stub();
+    };
+
+    const handleClickSpy = cy.spy(handleClick).as('handleClickSpy');
+
+    cy.mount(nonExpandableActionBar({ onClick: handleClickSpy }));
+
+    cy.get('[data-testid="action-bar"]').click();
+    cy.get('@handleClickSpy').should('have.been.called');
```

Comment:
You're right with that this test doesn't test what it should. However I really want non expandable action bar to have the opportunity to have an onClick after all as that function doesn't necessarily need to be used to open the accordion, it can be used for other useful stuff in future

Reply:
In that case, I guess all that's needed is to change the name of the test 😊

Reply:
Already did